doc/relevancy.md

```diff
@@ -0,0 +1,442 @@
+| string_for_sorting      | ✖                  | Used for storing a sortable subfield                                    | ✔                | ✔            | keyword                  | ✔    | ✔         | ✖            | ✖                   | ✖                 | ✖                   | ✖                            | ✖                | ✖               | ✖             | ✖                 |
+
+
+The main analyzer used at the moment is called query_with_old_synonyms.
+This is what all of the query components are using except for the part that
+attempts to use shingles.
+
+These are the steps, ignoring asciifolding, which we're adding at the moment:
+
+### 1) Normalise and strip quotes
+
+```
+normalize_quotes:
+  type: "mapping"
+  mappings:
+     - "\u0091=>\u0027"
+     - "\u0092=>\u0027"
+     - "\u2018=>\u0027"
+     - "\u2019=>\u0027"
+     - "\uFF07=>\u0027"
+
+strip_quotes:
+  type: "pattern_replace"
+  pattern: "\'"
+  replacement: ""
+```
+
+### 2) Split into tokens
+
+The standard tokeniser uses the unicode text segmentation algorithm.
+https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-standard-tokenizer.html
+
+### 3) Lowercase everything
+https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-lowercase-tokenfilter.html
+
+### 4) Remove stopwords
+https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-stop-tokenfilter.html
+
+Removes words in the english stopword list
+
+### 5) Apply our custom stemmer override to each token
+
+Customises stemming using this list
+https://github.com/alphagov/rummager/blob/master/config/schema/stems.yml
+
+### 6) Apply the english Stemmer to each token
+
+https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-stemmer-tokenfilter.html
+The Porter2 stemmer for english text
+
+### Recommendations
+
+1. Share custom stemming list with James
+2. Show how queries are analyzed in search admin or search performance explorer
+so that anyone can identify stemming problems and update the override list
+
+
+## ID codes
+
+Must query component of an unquoted search. E.g P 46 or (AB) 101 get tokenized to p46, ab101 and 101.
+
+Applies the following analysers on top of downcasing and asciifolding:
+
+- id_codes_find_phrases
+- id_codes_strip_symbols
+- id_codes_squash_multiple_spaces
+- id_codes_min_length
+- id_codes_make_groups
+- id_codes_strip_spaces
+- id_codes_require_digit
+- id_codes_min_length
+
+Which basically finds "codes" which consist of some letters and some digits,
+possibly split into two by whitespace. Requires there to be at least one
+digit and a minimum length of 2 characters.
+
+This does mean that if someone was to search for 'P45 and P46' it is possible
+to end up with a token of 'p45and'.
+
+
+## Long queries
```

Comment:
I don't think we've tried any of these.

Reply:
Should the Long queries section be removed?

Reply:
We should probably note them down somewhere as things we could try, but they don't really belong in this doc

Reply:
Same with weighting the top of the page more and a clean slate approach